src/vault.py

```diff
@@ -22,8 +22,16 @@ class VaultError(Exception):
 class Vault:
     """Class to interact with Vault through its API."""
 
-    def __init__(self, url: str, ca_cert_path: str):
-        self._client = hvac.Client(url=url, verify=ca_cert_path)
+    def __init__(self, url: str, auth_details: Dict[str, str]):
+        if ca_cert := auth_details.get("ca-cert-path"):
+            if (cert := auth_details.get("cert-path")) and (pk := auth_details.get("key-path")):
+                self._client = hvac.Client(url=url, cert=(cert, pk), verify=ca_cert)
+            else:
+                self._client = hvac.Client(url=url, verify=ca_cert)
+        else:
+            raise VaultError(
+                "The authentication provided to the vault client was incomplete or wrong."
```

Comment:
This message could be a bit clearer, as right now the only way this will be raised is if the `ca-cert-path` is not provided.

Reply:
Changed.